src/mpid/ch4/netmod/ofi/ofi_am_impl.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#include "mpidimpl.h"
+#include "ofi_impl.h"
+#include "ofi_am_impl.h"
+#include "ofi_noinline.h"
+
+
+int MPIDI_OFI_deferred_am_isend_issue(MPIDI_OFI_deferred_am_isend_req_t * dreq, int *status)
+{
+    int mpi_errno = MPI_SUCCESS;
+    char *send_buf;
+    MPI_Aint last;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_OFI_DEFERRED_AM_ISEND_ISSUE);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_OFI_DEFERRED_AM_ISEND_ISSUE);
+
+    MPIDU_genq_private_pool_alloc_cell(MPIDI_OFI_global.am_pack_buf_pool, (void **) &send_buf);
+    if (send_buf == NULL) {
+        /* TODO: extend deferred op for pipeline */
+        *status = MPIDI_OFI_OP_DEFERRED;
+        goto fn_exit;
+    }
+    mpi_errno = MPIR_Typerep_pack(dreq->buf, dreq->count_left, dreq->datatype, 0, send_buf,
+                                  dreq->data_sz, &last);
+    MPIR_ERR_CHECK(mpi_errno);
+
+    MPIDI_OFI_AMREQUEST_HDR(dreq->sreq, pack_buffer) = send_buf;
+
+    if (dreq->am_hdr_sz + dreq->data_sz + sizeof(MPIDI_OFI_am_header_t) <=
+        MPIDI_OFI_DEFAULT_SHORT_SEND_SIZE) {
+        mpi_errno =
+            MPIDI_OFI_am_isend_short(dreq->rank, dreq->comm, dreq->handler_id,
+                                     MPIDI_OFI_AMREQUEST_HDR(dreq->sreq, am_hdr), dreq->am_hdr_sz,
+                                     send_buf, dreq->data_sz, dreq->sreq);
+    } else {
+        mpi_errno =
+            MPIDI_OFI_am_isend_long(dreq->rank, dreq->comm, dreq->handler_id,
+                                    MPIDI_OFI_AMREQUEST_HDR(dreq->sreq, am_hdr), dreq->am_hdr_sz,
+                                    send_buf, dreq->data_sz, dreq->sreq);
+    }
+    MPIR_ERR_CHECK(mpi_errno);
+    *status = MPI_SUCCESS;
```

Comment:
`status` should not be set to an MPI error code in one place and to an enum in another place.